sdk/servicebus/azure-servicebus/azure/servicebus/_common/_configuration.py

```diff
@@ -30,3 +31,24 @@ def __init__(self, **kwargs):
         self.auto_reconnect = kwargs.get("auto_reconnect", True)
         self.keep_alive = kwargs.get("keep_alive", 30)
         self.timeout = kwargs.get("timeout", 60)  # type: float
+
+        self.custom_endpoint_address = kwargs.get("custom_endpoint_address")  # type: Optional[str]
+        self.connection_verify = kwargs.get("connection_verify")  # type: Optional[str]
+        self.connection_port = DEFAULT_AMQPS_PORT
+        self.custom_endpoint_hostname = None
+
+        if self.http_proxy or self.transport_type == TransportType.AmqpOverWebsocket:
+            self.transport_type = TransportType.AmqpOverWebsocket
+            self.connection_port = DEFAULT_AMQP_WSS_PORT
+
+        # custom end point
+        if self.custom_endpoint_address:
+            # if the custom_endpoint_address doesn't include the schema,
+            # we prepend a default one to make urlparse work
+            if self.custom_endpoint_address.find("//") == -1:
+                self.custom_endpoint_address = "sb://" + self.custom_endpoint_address
+            endpoint = urlparse(self.custom_endpoint_address)
+            self.transport_type = TransportType.AmqpOverWebsocket
```

Comment:
If someone has explicitly set the transport type to AMQP (as in, actually passed in that value, rather than just the default), and also supplied a custom_endpoint_address - we might want to raise a ValueError

Reply:
Discussed OOB - we'll address this separately - as it may also apply to http_proxy if the change is non-breaking.